docs/source/user_guide/best-practices-file-based-nodes.md

```diff
@@ -0,0 +1,102 @@
+<!--
+{% comment %}
+Copyright 2018-2021 Elyra Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+## Best practices for file-based pipeline nodes
+
+Generic pipelines and runtime specific pipelines support natively file-based nodes for  Jupyter notebooks, Python scripts, and R scripts. In order to support heterogeneous execution - that is making them runnable in any runtime environment (JupyterLab, Kubeflow Pipelines, and Apache Airflow) - follow the guidelines listed below.
+
+### Container image
+
+On Kubeflow Pipelines and Apache Airflow notebooks and scripts are executed in containers. Elyra ships with a few example container images to get you started, but you should consider utilizing purpose-built [container images](../recipes/creating-a-custom-runtime-image.md) instead. If possible, pre-install all software prerequisites in the container image you are using instead of installing them on the fly (e.g. by running `pip install my-package==1.2.3` in a notebook cell).
+
+### File I/O 
+
+In runtime environments (like Kubeflow Pipelines and Apache Airflow) where containers are used to run notebooks/scripts special consideration must be given to file input and output operations.
+
+#### File input
+
+If a notebook/script requires access to files that are stored on you local system, those files must be declared as input dependencies. Elyra collects declared files and uploads them to cloud storage and makes them available to the notebook/script at runtime.
+
+![Define input files](../images/user_guide/vpe-node-input-files.png)
+
+Input dependencies must be located in the pipeline file directory or a subdirectory of that location. Symlinks can be used to avoid the need to maintain multiple copies if files are shared among pipelines.
```

Comment:
A node dependency can only be a child of where the node file is located and it's related to the pipeline

Reply:
Fixed